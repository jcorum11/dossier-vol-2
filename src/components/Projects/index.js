import React from "react";

function Projects() {
  return (
    <section>
      <h1 data-testid="project-title" className="display-1 text-center">
        Projects
      </h1>
      <div className="container">
        <div className="row">
          <a
            className="col-sm rounded parent"
            href="https://johnnielsen1221.github.io/dd-initiative-tracker/"
          >
            <div id="proj-pic-1" className="rounded child">
              <span>D&D Initiative Tracker</span>
            </div>
          </a>
          <a
            className="col-sm rounded parent"
            href="https://obscure-reef-91673.herokuapp.com/"
          >
            <div id="proj-pic-2" className="rounded child">
              <span>Swap-Squad</span>
            </div>
          </a>
          <a
            className="col-sm rounded parent"
            href="https://rolodeck.herokuapp.com/"
          >
            <div id="proj-pic-rolodeck" className="rounded child">
              <span>RoloDeck</span>
            </div>
          </a>
        </div>
        <div className="row">
          <a
            className="col-sm rounded parent"
            href="https://jcorum11.github.io/run-buddy/"
          >
            <div id="proj-pic-3" className="rounded child">
              <span>Run Buddy</span>
            </div>
          </a>
          <a
            className="col-sm rounded parent"
            href="https://github.com/jcorum11/README.zing"
          >
            <div id="proj-pic-4" className="rounded child">
              <span>README.zing</span>
            </div>
          </a>
          <a
            className="col-sm rounded parent"
            href="https://jcorum11.github.io/test-rabbit/"
          >
            <div id="proj-pic-5" className="rounded child">
              <span>Test Rabbit</span>
            </div>
          </a>
        </div>
      </div>
    </section>
  );
}

export default Projects;
