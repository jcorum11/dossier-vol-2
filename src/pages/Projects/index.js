import SectionTitle from '../../components/SectionTitle'
import styled from 'styled-components'
import Project from '../../components/Project'

function Projects() {
  const projectDetails = {
    dnd: {
      description: 'This was a really cool tracker that I did during bootcamp. Check it out!',
      href: 'https://johnnielsen1221.github.io/dd-initiative-tracker/'
    },
    rolodeck: {
      description: 'I am really proud of this one. I got a group of friends together and we worked really hard on it. I really loved working with them. You should definitely check this one out.',
      href: 'https://obscure-reef-91673.herokuapp.com/'
    },
    harpoon: {
      description: 'I really love baseball cards. Like... I really love them. I made an app in React Native for my friend as a kind of social media for baseball cards. This one doesnt have a link yet but I hope it will soon!',
      href: ''
    },
    sploots: {
      description: 'This is another one that I did with a friend of mine. It too doesnt have a link because we are still working on it, but it is a website for people to sell dogs and pet services that has a really good UI.',
      href: ''
    }
  }
  return (
    <section id="projects" className="py-5">
      <SectionTitle name='projects' />
      <section>
        <Project title='Harpoon Mobile' description={projectDetails.harpoon.description} href={projectDetails.harpoon.href} first={true}/>
        <Project title='Sploots!' description={projectDetails.sploots.description} href={projectDetails.sploots.href} />
        <Project title='Rolodeck' description={projectDetails.rolodeck.description} href={projectDetails.rolodeck.href} />
        <Project title='D&D Tracker' description={projectDetails.dnd.description} href={projectDetails.dnd.href} />
      </section>
    </section>
  )
}

export default Projects
