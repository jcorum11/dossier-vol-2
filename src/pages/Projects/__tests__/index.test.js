import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Projects from "..";

afterEach(cleanup);

describe("Projects component", () => {
  it("renders", () => {
    render(<Projects />);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Projects />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("inserts header correctly", () => {
  it("says 'Projects' at the top", () => {
    const { getByTestId } = render(<Projects />);
    expect(getByTestId("project-title")).toHaveTextContent("Projects");
  });
});
