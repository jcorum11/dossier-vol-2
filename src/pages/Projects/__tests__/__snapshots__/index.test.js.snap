// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects component matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h1
      data-testid="project-title"
    >
      Projects
    </h1>
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-sm"
          id="proj-pic-1"
        />
        <div
          class="col-sm"
          id="proj-pic-2"
        />
        <div
          class="col-sm"
          id="proj-pic-3"
        />
      </div>
      <div
        class="row"
      >
        <div
          class="col-sm"
          id="proj-pic-4"
        />
        <div
          class="col-sm"
          id="proj-pic-5"
        />
        <div
          class="col-sm"
          id="proj-pic-6"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
